### ISSUE TITLE
User record conflict between LDAP and the local DB

### ISSUE THREAD

**Issue by morungos** [capsid/capsid-webapp#22]
CaPSID seems to use both a local DB record and an LDAP record for identifying users. However, it's broken when the user exists in LDAP but not in the local DB, rendering errors when it can't locate the local user record. We should either (a) not let a user log in if there isn't a user record, or (b) ensure that we create a user record as needed. In any event, the current behaviour (deep rendering errors in the Bootstrap template) is not a reasonable way forward.

**Comment by morungos** [comment]
Typical manifestation in the log is:

```
Caused by: org.codehaus.groovy.grails.web.pages.exceptions.GroovyPagesException: Error processing GroovyPageView: Error executing tag <sec:ifLoggedIn>: Cannot get property 'username' on null object
    at org.codehaus.groovy.grails.web.servlet.view.GroovyPageView.createGroovyPageException(GroovyPageView.java:205)
    at org.codehaus.groovy.grails.web.servlet.view.GroovyPageView.handleException(GroovyPageView.java:181)
    at org.codehaus.groovy.grails.web.servlet.view.GroovyPageView.renderWithTemplateEngine(GroovyPageView.java:152)
    at org.codehaus.groovy.grails.web.servlet.view.GroovyPageView.renderMergedOutputModel(GroovyPageView.java:83)
    at org.springframework.web.servlet.view.AbstractView.render(AbstractView.java:262)
    at org.codehaus.groovy.grails.web.sitemesh.SpringMVCViewDecorator.render(SpringMVCViewDecorator.java:67)
    ... 59 more
```

Note "property" not method. These appear to be references such as `id="${user.username}"`, where the `user` is retrieved but only assumed to exist. Same for the bookmarks. These would stop the page crashing. The login process issue is probably best handled with a separate ticket.

**Comment by morungos** [comment]
Resolved in e676495
